joblib/memory.py

```diff
@@ -752,6 +752,48 @@ class Memory(Logger):
         structure.
 
         Read more in the :ref:`User Guide <memory>`.
+
+        Parameters
+        ----------
+        location: str or None
+            The path of the base directory to use as a data store
+            or None. If None is given, no caching is done and
+            the Memory object is completely transparent. This option
+            replaces cachedir since version 0.12.
+
+        backend: str or 'local'
```

Comment:
A bit orthogonal to moving the docstring around, but you may as well just use `str` here. `str or 'local'` is slightly weird since `'local'` is a `str`.

I am guessing that a few things are optional and not marked as such.